src/app/wallets/index.types.d.ts

```diff
@@ -0,0 +1,27 @@
+type AddInvoiceSelfArgs = {
+  walletId: WalletId
+  amount: Satoshis
+  memo?: string
+}
+
+type AddInvoiceNoAmountSelfArgs = {
+  walletId: WalletId
+  memo?: string
+}
+
+type AddInvoiceRecipientArgs = {
+  username: Username
+  amount: Satoshis
+  memo?: string
+}
+
+type AddInvoiceNoAmountRecipientArgs = {
+  username: Username
+  memo?: string
+}
+
+type BaseAddInvoiceArgs = {
```

Comment:
I wouldn't declare this here as it is for a function that _isn't_ exposed publicly. Just declare the arg types inline in the function definition. Only declare types here that need exporting.